### ISSUE TITLE
type bounds at runtime

### ISSUE THREAD

**Issue by trufae** [vlang/v#5915]
**V version:** V 0.1.28 e1eb9c4
**OS:** macOS 10.15

**What did you do?**

```go
a := byte(10)
a += 3200
```
**What did you expect to see?**

runtime error, because the byte variable `a` overflows. I would suggest making the safe overflow checks by default, but disable them in case that +, - or * operations are done inside an `unsafe`block like this:

```go
a := byte(10)
unsafe {
  a += 3200 // no runtime assert
}
```

additionally it will be good to add a way to check for overflow, i can think on having methods to check this on every type, and also a try/catch model, but this will require more discussion:

```go
a := byte(10)
a += 3200 or {
   panic('overflow')
}
```

Another option is to create safe int types. like it's done in some c++ templates like this:

a := safe_byte(10)
if a.safe_add(3200) {
  a += 3200
} else {
  panic('overflow')
}

**What did you see instead?**

nothing happens. program behaves in the unexpected (or expected, because we are used to this) integer overflowy way.

**Comment by ntrel** [comment]
`+= 3200` should be a compile-time error on a byte. 

Sometimes overflow/underflow is useful, and V is a system language. I think modern hardware has instructions to trap underflow/overflow, that's probably the most efficient way to detect it.

`unsafe` is for memory unsafe code. Overflow/underflow is memory safe.

**Comment by ArtemkaKun** [comment]
Hi @trufae, maybe we should have this as *Feature request* instead of *Bug*, what do you think?

**Comment by trufae** [comment]
Yep, this sounds more like a design decision and something that must be discussed as a festure rather than a bug because most languages handle numbers the way v does already. How can i change the ticket? Do i need to open a new one?
